mobile/lib/pages/search/search.page.dart

```diff
@@ -441,19 +441,15 @@ class SearchPage extends HookConsumerWidget {
     }
 
     handleTextSubmitted(String value) {
-      if (value.isEmpty) {
-        return;
-      }
-
       if (isContextualSearch.value) {
         filter.value = filter.value.copyWith(
-          filename: null,
```

Comment:
if you don't send null, the search will be incorrect

Reply:
why?

Reply:
https://github.com/immich-app/immich/blob/c3be74c450ba7d57ccfc9f6c0463478e579682a0/mobile/lib/services/search.service.dart#L82-L85

Reply:
What are you reference there?

Reply:
where the filename filter is being used

Reply:
It is used when it is not null and not an empty string

Reply:
Also, context and file name search are two different search mechanisms, they are not overlapping. It is the reason we have to set null here

Reply:
Am I missing something here? By setting the filename or the context to empty string we essentially make a search request without them which is the correct thing to do.

Reply:
and the empty string is necessary because if you copyWith null you will get the previous value

Reply:
They should be set null. I'm not sure if the server treats an empty string like null in this case, and in general a lack of a value should be represented with null instead of an empty string.

Reply:
@mertalev the server doesn't get an empty string, it gets null, read the referenced code

Reply:
that's what I mean when I said 

> The SearchService handles it by checking if the strings context and filename are empty.

Reply:
Oh, I see. It seems like a weird limitation to not be able to change an earlier value to null. Not sure if there's a better way to handle that, but I guess this is okay for now if there isn't.